haskell-src/Concordium/Types/InvokeContract.hs

```diff
@@ -0,0 +1,115 @@
+-- |Types related to simulating contract invocations.
+
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE NumericUnderscores #-}
+module Concordium.Types.InvokeContract
+  ( ContractContext(..)
+  , InvokeContractResult(..)
+  ) where
+
+import qualified Data.Aeson as AE
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Base16 as BS16
+import qualified Data.Text.Encoding as Text
+
+
+import qualified Concordium.Wasm as Wasm
+import Concordium.Types (Address, Amount, ContractAddress, Energy)
+import Concordium.Types.Execution (Event, RejectReason)
+
+data ContractContext = ContractContext {
+  -- |Invoker of the contract. If this is not supplied then the contract will be
+  -- invoked, by an account with address 0, no credentials and sufficient amount
+  -- of CCD to cover the transfer amount. If given, the relevant address must
+  -- exist in the blockstate.
+  ccInvoker :: !(Maybe Address),
+  -- |Contract to invoke.
+  ccContract :: !ContractAddress,
+  -- |Amount to invoke the contract with.
+  ccAmount :: !Amount,
+  -- |Which entrypoint to invoke.
+  ccMethod :: !Wasm.ReceiveName,
```

Comment:
Could/should this be a `Wasm.EntrypointName` instead?

Reply:
For consistency with "Update" I prefer to use ReceiveName,